Healthy/Classes/Utilities/Authentication/AuthenticationProtocol.swift

```diff
@@ -0,0 +1,7 @@
+import UIKit
+
+public struct AuthenticatedUser {}
+
+ public protocol Authentication {
+     func performLogin(withPresenting viewController: UIViewController) async throws -> AuthenticatedUser
```

Comment:
```suggestion
     func performLogin() async throws -> AuthenticatedUser
```

What if I need to use this authenticator in SwiftUI? Please avoid adding presentation logic to business logic.

Reply:
How would I instantiate signIn(withPresenting:)? should I make a property of type UIViewController?